L1Trigger/TrackerTFP/src/GeometricProcessor.cc

```diff
@@ -0,0 +1,126 @@
+#include "L1Trigger/TrackerTFP/interface/GeometricProcessor.h"
+
+#include <numeric>
+#include <algorithm>
+#include <iterator>
+#include <deque>
+#include <vector>
+
+using namespace std;
+using namespace edm;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  GeometricProcessor::GeometricProcessor(const ParameterSet& iConfig, const Setup* setup, const DataFormats* dataFormats, int region) :
+    enableTruncation_(iConfig.getParameter<bool>("EnableTruncation")),
+    setup_(setup),
+    dataFormats_(dataFormats),
+    region_(region),
+    input_(dataFormats_->numChannel(Process::gp), vector<deque<StubPP*>>(dataFormats_->numChannel(Process::pp))) {}
+
+  void GeometricProcessor::consume(const TTDTC& ttDTC) {
+    auto validFrame = [](int& sum, const TTDTC::Frame& frame){ return sum += frame.first.isNonnull() ? 1 : 0; };
```

Comment:
// Add comment "Read in & organise input data (fill array input_)"

Reply:
done